src/raygun.rum/index.js

```diff
@@ -958,6 +984,82 @@ var raygunRumFactory = function(window, $, Raygun) {
       makePostCorsRequest(url, stringifiedPayload, successCallback, errorCallback);
     }
 
+     function updateUserAgentData(payload) {
+       if (!payload.eventData) { return; }
+
+       for (var i = 0; i < payload.eventData.length; i++) {
+         if (!!payload.eventData[i].data && Array.isArray(payload.eventData[i].data)) {
+
+           if (!!payload.eventData[i].device) {
+             payload.eventData[i].device = getHighFidelityUAString(payload.eventData[i].device);
+           }
+
+           for (var k = 0; k < payload.eventData[i].data.length; k++) {
+             var dataFragment = payload.eventData[i].data[k];
+
+             if (!!dataFragment.userAgent) {
+              dataFragment.userAgent = getHighFidelityUAString(dataFragment.userAgent);
+             }
+
+             if (!!window.raygunUserAgentData && !!window.raygunUserAgentData.platformVersion) {
+               var platformVersion = (window.raygunUserAgentData.platformVersion || '.').split(".");
+               dataFragment.device = {
+                 Family: window.raygunUserAgentData.platform,
+                 Major: platformVersion[0] || '',
+                 Minor: platformVersion[1] || '',
+                 Patch: platformVersion[2] || '',
+                 PatchMinor: platformVersion[3] || ''
+               };
```

Comment:
Instead of `(window.raygunUserAgentData.platformVersion || '.').split(".")` I'd usually opt for `(window.raygunUserAgentData.platformVersion || '').split(".")`, just to keep things simple.

Is it important that we have an array with two empty strings if `window.raygunUserAgentData.platformVersion` is undefined?

Reply:
No. just me over thinking things